### PR TITLE
API Changes: HTTP

### PR DESCRIPTION
**Pull request by amandawrcoelho**
Rename some functions

Functions sol_http_params_add() and sol_http_params_add_copy() now
return an int.

issue: #1717

Signed-off-by: Amanda Coelho <amada.winder.ribeiro.coelho@intel.com>

### CODE REVIEW

**Comment by otaviobp** [comment]
lgtm

File: src/samples/http/client.c [hunk 1]
```diff
@@ -169,7 +169,7 @@ create_header_params(struct sol_http_params *params, const char *value)
     param.value.key_value.key = key;
     param.value.key_value.value = val;
 
-    if (!sol_http_param_add_copy(params, param)) {
+    if (sol_http_params_add_copy(params, param) < 0) {
```
Comment (glima) [inline src/samples/http/client.c]: On all those bool->int conversions where we got also logging, a ``sol_util_strerrora()`` usage won't hurt :)

Comment (ceolin) [inline src/samples/http/client.c]: this can be done further

**Comment by ceolin** [comment]
it is in, thanks !

**Comment by amandawrcoelho** [comment]
tks for all the reviews :)

**Pull request closed without merging**